### DISCUSSION TITLE
i3 with nvidia does not start

### DISCUSSION THREAD

**Discussion by pasbi** [i3/i3#6228] [Q&A]
I have a laptop with integrated Intel GPU and nvidia (ada RTX500) GPU and I use [`envycontrol`](https://github.com/bayasdev/envycontrol) to switch between them.

When I select the nvidia GPU (`envycontrol -s nvidia`) and reboot, i3 does not start correctly anymore (gnome does).
My `.xinitrc`:

```
#!/bin/sh

# xrandr --setprovideroutputsource modesetting NVIDIA-0
# xrandr --auto

exec i3 -d all -V > i3log

#exec /usr/bin/gnome-session  # gnome starts properly
```

Enabling the `xrandr`-lines does not help, same symptom: `startx` prints something, then the screen gets black. I am able to recover by going to another tty (e.g., CTRL+ALT+F5) and `killall i3`.

[This is the log](https://gist.github.com/pasbi/dd774784441f387e2c81ada15993a814).

For debugging I manually run `startx` from a tty, usually I use `gdm`, both show the same symptoms.
I have a virgin i3 config.
After switching to the Intel GPU (`envycontrol -s integrated`), both i3 and gnome work as expected.

Why doesn't `i3` work with the nvidia GPU?

**Comment by Certainty1396** [comment]
I switch from gnome to i3wm. I just append one line to config file of i3
exec xrandr --auto
All is normal.

**Reply by pasbi** [reply]
Thanks for your answer!
Indeed I should have put `exec xrandr --auto` into `.config/i3/config`, not in `.xinitrc` -- my bad.
And it works if I replace it with `exec xrandr --output eDP-1`, the `xrandr --auto` does not work.
I guess that's not a i3-specific error, so it's off-topic here.

But I wonder why i3 needs this line while gnome works fine without (`xrandr --auto` doesn't work with gnome either, but I don't need it there)?
